src/sysfiles.c

```diff
@@ -84,15 +84,27 @@ typedef void sig_handler_t ( int );
 #include <mach-o/dyld.h>
 #endif
 
+#include <zlib.h>
+
 
 /* utility to check return value of 'write'  */
 ssize_t echoandcheck(int fid, const char *buf, size_t count) {
   int ret;
+  if (syBuf[fid].type == gzip_socket) {
+      ret = gzwrite(syBuf[fid].gzfp, buf, count);
+      if (ret < 0)
+          ErrorQuit(
+              "Cannot write to compressed file, see 'LastSystemError();'\n",
```

Comment:
Language nitpick: If I saw that message, I would think "ah, writing to compressed files is not supported"; but I think what really is meant here is that writing to a compressed file failed. Perhaps "Could not write" or "Failed to write" would be clearer (or maybe that's me as a non-native speaker just not getting it)?

(Again, not necessarily asking you to change this, just wondering)